api/app/c_haines/c_haines_index.py

```diff
@@ -0,0 +1,182 @@
+""" Logic pertaining to the generation of c_haines index from GDAL datasets.
+"""
+import logging
+import struct
+from typing import Final, Tuple
+from pyproj import CRS
+from osgeo import gdal
+from app.weather_models.process_grib import (
+    calculate_geographic_coordinate, get_dataset_geometry, get_transformer, GEO_CRS)
+from app.c_haines import GDALData
+
+
+logger = logging.getLogger(__name__)
+
+
+def calculate_c_haines_index(t700: float, t850: float, d850: float) -> float:
+    """ Given temperature and dew points values, calculate c-haines.
+
+    Parameters
+    ----------
+    t700 : float
+        Temperature at 700 mb.
+    t850 : float
+        Temperature at 850 mb.
+    d850 :float
+        Dew point depression at 850mb.
+
+    Returns
+    -------
+    float
+        The Continous Haines Index.
+    """
+    # pylint: disable=invalid-name
+
+    # Temperature depression term (this indicates atmospheric instability).
+    # Temperature at 850mb - Temperature at 700mb.
+    ca = (t850-t700)/2-2
+    # Dew point depression term (this indicates how dry the air is).
+    cb = d850/3-1
+
+    # This part limits the extent to which dry air is able to affect the overall index.
+    # If there is very dry air (big difference between dew point temperature and temperature),
+    # we want to limit the overall effect on the index, since if there's no atmospheric
+    # instability, that dry air isn't going anywhere.
+    if cb > 9:
+        cb = 9
+    elif cb > 5:
+        cb = 5 + (cb-5)/2
+
+    # Combine the two terms for the index.
+    ch = ca + cb
+
+    return ch
```

Comment:
@andrea-williams super good point Andrea. I've added citations. I've been going off random things emailed to me and found on the internet thus far. This is a (small) can of worms, since our code doesn't exactly match the source.